src/fides/api/ops/service/connectors/saas_connector.py

```diff
@@ -116,7 +116,16 @@ def retrieve_data(
 
         query_config: SaaSQueryConfig = self.query_config(node)
         read_request: Optional[SaaSRequest] = query_config.get_request_by_action("read")
+        delete_request: Optional[SaaSRequest] = query_config.get_request_by_action(
+            "delete"
+        )
+
         if not read_request:
+            # if a delete request is specified for this endpoint without a read request
+            # then we return a single empty row to still trigger the mask_data method
+            if delete_request:
+                return [{}]
```

Comment:
Would it be worth logging this to the terminal too?

Reply:
Log message added